### ISSUE TITLE
Changing `exit_condition` in the `WindowPlugin` from the default cause panic on startup.

### ISSUE THREAD

**Issue by Neopallium** [bevyengine/bevy#8057]
## Bevy version

0.10.0

## What you did

I was able to reproduce the bug using the `https://github.com/bevyengine/bevy/blob/latest/examples/window/scale_factor_override.rs` example.  Unmodified the example runs correctly on my system.  But if the `exit_condition` is changed from it's default value, the example will panic on startup.

```rust
        .add_plugins(DefaultPlugins.set(WindowPlugin {
            primary_window: Some(Window {
                resolution: WindowResolution::new(500., 300.).with_scale_factor_override(1.0),
                ..default()
            }),
            // Either of these exit conditions will cause the panic.
            exit_condition: bevy::window::ExitCondition::OnPrimaryClosed,
            //exit_condition: bevy::window::ExitCondition::DontExit,
            ..default()
        }))
```

## What went wrong

Panic when bevy is building the schedule:
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', crates/bevy_ecs/src/schedule/schedule.rs:1146:56
stack backtrace:
   0: rust_begin_unwind
             at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/panicking.rs:64:14
   2: core::panicking::panic
             at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/panicking.rs:111:5
   3: core::option::Option<T>::unwrap
             at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/option.rs:778:21
   4: bevy_ecs::schedule::schedule::ScheduleGraph::build_schedule
             at ./crates/bevy_ecs/src/schedule/schedule.rs:1146:34
   5: bevy_ecs::schedule::schedule::ScheduleGraph::update_schedule
             at ./crates/bevy_ecs/src/schedule/schedule.rs:1326:21
   6: bevy_ecs::schedule::schedule::Schedule::initialize
             at ./crates/bevy_ecs/src/schedule/schedule.rs:259:13
   7: bevy_ecs::schedule::schedule::Schedule::run
             at ./crates/bevy_ecs/src/schedule/schedule.rs:248:9
   8: bevy_ecs::world::World::run_schedule_ref
             at ./crates/bevy_ecs/src/world/mod.rs:1748:9
   9: bevy_ecs::world::World::run_schedule
             at ./crates/bevy_ecs/src/world/mod.rs:1724:9
  10: bevy_app::CoreSchedule::outer_loop
             at ./crates/bevy_app/src/lib.rs:75:9
```

## Additional information

I think it is cause because the `exit_on_all_closed` system is not used for those two conditions, but this code references that system:
https://github.com/bevyengine/bevy/blob/289fd1d0f2353353f565989a2296ed1b442e00bc/crates/bevy_winit/src/lib.rs#L80

**Comment by mockersf** [comment]
should be fixed by #7950

**Comment by Neopallium** [comment]
Confirmed fixed.  Nice work. @mockersf
